client/network/src/protocol/sync.rs

```diff
@@ -1156,60 +1241,210 @@ impl<B: BlockT> ChainSync<B> {
 		self.pending_requests.set_all();
 	}
 
-	/// Call when a node announces a new block.
+	/// Checks if there is a slot for a block announce validation.
 	///
-	/// If `OnBlockAnnounce::ImportHeader` is returned, then the caller MUST try to import passed
-	/// header (call `on_block_data`). The network request isn't sent
-	/// in this case. Both hash and header is passed as an optimization
-	/// to avoid rehashing the header.
-	pub fn on_block_announce(&mut self, who: &PeerId, hash: B::Hash, announce: &BlockAnnounce<B::Header>, is_best: bool)
-		-> OnBlockAnnounce
-	{
+	/// The total number and the number per peer of concurrent block announce validations
+	/// is caped.
+	///
+	/// It will return `true` when there is still a slot for a block announce validation.
```

Comment:
It doesn't return a boolean. I guess this is from before `HasSlotForBlockAnnounceValidation` was introduced?

Reply:
Good find :)